build.rs

```diff
@@ -1,6 +1,30 @@
 use std::env;
 
+include!("src/dist/triple.rs");
+
+pub fn from_build() -> Result<PartialTargetTriple, String> {
+    if let Some(triple) = env::var("RUSTUP_OVERRIDE_BUILD_TRIPLE").ok() {
+        PartialTargetTriple::new(&triple).ok_or(triple)
+    } else {
+        let triple = env::var("TARGET").unwrap();
+        PartialTargetTriple::new(&triple).ok_or(triple)
+    }
+}
+
 fn main() {
+    println!("cargo:rerun-if-env-changed=RUSTUP_OVERRIDE_BUILD_TRIPLE");
+    println!("cargo:rerun-if-env-changed=TARGET");
+    match from_build() {
+        Ok(triple) => eprintln!("Computed build based partial target triple: {:#?}", triple),
+        Err(s) => {
+            eprintln!("Unable to parse target '{}' as a PartialTargetTriple", s);
+            eprintln!(
+                "If you are attempting to bootstrap a new target you may need to adjust the\n\
+               permitted values found in src/dist/triple.rs"
+            );
+            std::process::abort();
```

Comment:
Hmm, is there any reason you chose to abort instead of exiting?
```suggestion
            std::process::exit(1);
```

Reply:
I chose to abort because I wasn't *certain* that non-zero was correct on all platforms.

I can change it if you really want, but this seemed easiest/clearest.  I also considered just panic!()ing

Reply:
Hmm, ok. I guess rust doesn't have an EXIT_FAILURE like C does, I'm not used to thinking about portability :smile: